src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,177 @@
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>(tail, value, null);
+        Node<T> temp = tail;
+        temp.next = newNode;
+        tail = newNode;
+        if (head.next == null) {
+            head.next = temp;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index != size) {
+            rangeCheck(index);
+        }
+        if (head == null || head == tail) {
+            add(value);
+            return;
+        }
+        Node<T> node = getNode(index);
+        if (node == head) {
+            Node<T> newNode = new Node<>(null, value, node);
+            node.previous = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        if (node == tail && index == size) {
+            Node<T> newNode = new Node<>(node, value, null);
+            node.next = newNode;
+            tail = newNode;
+            size++;
+            return;
+        }
+        Node<T> newNode = new Node<>(node.previous, value, node);
+        node.previous.next = newNode;
+        node.previous = newNode;
+        if (node.next != null) {
+            node.next = node.next.previous;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeCheck(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        rangeCheck(index);
+        Node<T> node = getNode(index);
+        T ele = node.value;
+        node.value = value;
+        return ele;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeCheck(index);
+        Node<T> node = getNode(index);
+        return unlinkAndRemove(node);
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> tempNode = head;
+        for (int i = 0; i < size; i++) {
+            T value = tempNode.value;
+            if (object == null && value == null || value != null && value.equals(object)) {
+                unlinkAndRemove(tempNode);
+                return true;
+            }
+            tempNode = tempNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> getNode(int index) {
+        int middle = size >> 1;
+        if (index <= middle) {
+            Node<T> headEl = head;
+            for (int i = 0; i < index; i++) {
+                headEl = headEl.next;
+            }
+            return headEl;
+        }
+        Node<T> tailEl = tail;
+        for (int i = size - 1; i > index; i--) {
+            tailEl = tailEl.previous;
+        }
```

Comment:
in this method also better to use if - else. Don’t use abbreviations for variable names.